File: tests/fair_value_gap_test.py
import numpy as np

from taflow import FairValueGap


def test_fair_value_gap_lifecycle_and_reset():
    state = FairValueGap()
    state.extend([10.0, 10.5], [11.0, 12.0], [9.0, 10.0], [10.5, 11.5])
    assert len(state) == 2
    assert len(state.compute()) == 4
    state.reset()
    assert state.value is None

File: crates/taflow-core/src/indicators/fair_value_gap_test.rs
use super::fair_value_gap::FairValueGap;

#[test]
fn lifecycle_and_reset_are_consistent() {
    let mut state = FairValueGap::new();
    assert!(state.append(10.0, 11.0, 9.0, 10.5).is_some());
    state.reset();
    assert!(state.value().is_none());
}
